config/env.go

```diff
@@ -124,6 +131,18 @@ func FromEnv() HCPConfigOption {
 			config.scadaTLSConfig = scadaTLSConfig
 		}
 
+		// Read user profile information from the environment, the values will only be
+		// used if both fields are provided.
+		hcpOrganizationID, hcpOrganizationIDOK := os.LookupEnv(envVarHCPOrganizationID)
+		hcpProjectID, hcpProjectIDOK := os.LookupEnv(envVarHCPProjectID)
+
+		if hcpOrganizationIDOK && hcpProjectIDOK {
+			userProfile := profile.UserProfile{OrganizationID: hcpOrganizationID, ProjectID: hcpProjectID}
+			if err := apply(config, WithProfile(&userProfile)); err != nil {
+				return fmt.Errorf("failed to set profile fields from environment variables (%s, %s): %w", envVarHCPOrganizationID, envVarHCPProjectID, err)
```

Comment:
For consistency with the errors above:
```suggestion
				return fmt.Errorf("failed to configure profile based on environment variables (%s, %s): %w", envVarHCPOrganizationID, envVarHCPProjectID, err)
```

(bonus ask: fix the `basd` typo I just noticed in those other errors 😅)